fixups/EnvVarFixup/EnvVarFixup.cpp

```diff
@@ -0,0 +1,376 @@
+                                return result;
+                            }
+                            else
+                            {
+#if _DEBUG
+                                Log("[%d] GetEnvironmentVariableFixup:(W) HKCU value Failed 0x%x.", GetEnvVarInstance, ret00);
+#endif                               
+                            }
+
+                            RegCloseKey(hKeyLM);
+                        }
+                    }
+                    else
+                    {
+                        if (RegOpenKeyExW(HKEY_LOCAL_MACHINE, L"SYSTEM\\CurrentControlSet\\Control\\Session Manager\\Environment",
+                            0, KEY_ENUMERATE_SUB_KEYS | STANDARD_RIGHTS_READ | KEY_QUERY_VALUE, &hKeyLM) == ERROR_SUCCESS)
+                        {
+#if _DEBUG
+                            Log("[%d] GetEnvironmentVariableFixup:(W) HKLM key found.", GetEnvVarInstance);
+#endif                        
+                            DWORD type = RRF_RT_REG_SZ;
+                            DWORD dLen = lenBuf;
+                            auto ret = RegGetValueW(HKEY_LOCAL_MACHINE, L"SYSTEM\\CurrentControlSet\\Control\\Session Manager\\Environment", lpName,
+                                RRF_RT_REG_SZ | RRF_RT_REG_EXPAND_SZ | RRF_ZEROONFAILURE, &type, lpValue, &dLen);
+                            if (ret == ERROR_SUCCESS)
+                            {
+#if _DEBUG
+                                Log("[%d] GetEnvironmentVariableFixup:(W) HKLM value found %ls", GetEnvVarInstance, lpValue);
+#endif                        
+                                result = dLen;
+                                RegCloseKey(hKeyLM);
+                                return result;
+                            }
+                            else
+                            {
+#if _DEBUG
+                                Log("[%d] GetEnvironmentVariableFixup:(W) HKLM value Failed 0x%x.", GetEnvVarInstance, ret);
+#endif 
+                            }
+                            RegCloseKey(hKeyLM);
+                        }
+                    }
+                    // allow to fall through to system
+                    //return result;
+                    result = 0;
+                }
+
+                // Sometimes HKLM\System reg items from the package are not visible to the app.  We think this is a bug, so
+                // allow a check to see if there is a value field in the JSON and use that.
+                // Of course, we should always look at the json if useregistry was set to false anyway.
+#if _DEBUG
+                Log("[%d] GetEnvironmentVariableFixup: Json supplied case.", GetEnvVarInstance);
+#endif
+
+                if (valuelen < lenBuf)
+                {
+#if _DEBUG
+                    Log("[%d] GetEnvironmentVariableFixup: Match to be returned.", GetEnvVarInstance);
+#endif
+                    // TODO: copy into lpValue, but might need form conversion

```

Comment:
Can you please remove this TODO and add it to the list of issues for PSF?

Reply:
Done. Code had already been added so this was actually a TODONE.